### PR TITLE
Code Restructuring only, no change with respect to cmdlet behaviour

### PR DESCRIPTION
**Pull request by raviten**
…oss code base

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/DataBoxEdgeRoleTests.ps1 [hunk 1]
```diff
@@ -48,9 +48,8 @@ function Test-CreateRole
 	$iotDeviceConnectionString = Get-IotDeviceConnectionString
 	$iotDeviceConnSec = ConvertTo-SecureString $iotDeviceConnectionString -AsPlainText -Force
 
-	$encryptionKeyString = Get-EncryptionKey 
-	$encryptionKey = ConvertTo-SecureString $encryptionKeyString -AsPlainText -Force
-
+	$encryptionKey = Get-EncryptionKey
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/DataBoxEdgeRoleTests.ps1]: since your test case is in live only mode, how can you verify your change is valid?

File: src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1 [hunk 2]
```diff
@@ -48,7 +48,8 @@ Returns standard EncryptionKey
 #>
 function Get-EncryptionKey
 {
-	$encryptionKey = ConvertTo-SecureString -String ""
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1]: correct me if I was wrong, I think this line will fail immediately, and this test case is in live only mode so for your new changes, there are no way for CI to verify them.

File: src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1 [hunk 3]
```diff
@@ -48,7 +48,8 @@ Returns standard EncryptionKey
...
+	$val = Get-AzKeyVaultSecret -VaultName azpsdbe -Name "EncryptionKey"
+	$encryptionKey = $val.SecretValueText
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1]: why not return "$val.SecretValueText" directly

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleGetCmdlet.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
+﻿// ----------------------------------------------------------------------------------
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleGetCmdlet.cs]: this file and "src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidth/DataBoxEdgeBandwidthScheduleGetCmdletBase.cs"
doesn't look like the same, it's hard to tell what you changed.

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs [hunk 5]
```diff
@@ -12,32 +12,32 @@
...
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 using Microsoft.Azure.Management.EdgeGateway;
+using Microsoft.Azure.Management.EdgeGateway.Models;
+using Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models;
 using Microsoft.Rest.Azure;
 using Microsoft.WindowsAzure.Commands.Utilities.Common;
-using ResourceModel = Microsoft.Azure.Management.EdgeGateway.Models.BandwidthSchedule;
-using PSResourceModel = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeBandWidthSchedule;
-using Resource = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Resources.Resource;
+using System;
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs]: better not to make unnecessary changes like the order of using statements.

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs [hunk 6]
```diff
@@ -12,32 +12,32 @@
...
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Net;
 
-namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Common.Cmdlets.Bandwidth
+namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Common.Cmdlets.Bandwidths
 {
     [Cmdlet(VerbsCommon.New,
          Constants.BandwidthSchedule,
          DefaultParameterSetName = BandwidthParameterSet,
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs]: namespace and folder names are changed to "Bandwidths", should this parameter set also be "BandwidthsParameterSet"?

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs [hunk 7]
```diff
@@ -12,32 +12,32 @@
...
          SupportsShouldProcess = true),
-     OutputType(typeof(PSResourceModel))]
-    public class DataBoxEdgeBandwidthNewCmdletBase : AzureDataBoxEdgeCmdletBase
+     OutputType(typeof(PSDataBoxEdgeBandWidthSchedule))]
+    public class DataBoxEdgeBandwidthScheduleNewCmdlet : AzureDataBoxEdgeCmdletBase
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs]: should this be "DataBoxEdgeBandwidthsScheduleNewCmdlet"?

File: src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1 (hunk 2 above)
Comment (raviten) [reply src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1]: Yes, since this involves adding a secret in the code base, I need to migrate to azure keyvault. Currently, I am testing everythin locally by adding the encryption keys and running tests.

Removing the encryption key before making a commit

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs [hunk 8]
```diff
@@ -46,13 +46,13 @@ public class DataBoxEdgeBandwidthNewCmdletBase : AzureDataBoxEdgeCmdletBase
...
         [Parameter(Mandatory = true,
             HelpMessage = Constants.DeviceNameHelpMessage,
+            ValueFromPipelineByPropertyName = true,
             Position = 1)]
         [ResourceNameCompleter("Microsoft.DataBoxEdge/dataBoxEdgeDevices", nameof(ResourceGroupName))]
         [ValidateNotNullOrEmpty]
 
         public string DeviceName { get; set; }
 
-
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleNewCmdlet.cs]: probably white spaces

File: src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1 (hunk 3 above)
Comment (raviten) [reply src/DataBoxEdge/DataBoxEdge.Test/ScenarioTests/Common.ps1]: Sure, will do

File: src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleRemoveCmdlet.cs [hunk 9]
```diff
@@ -13,20 +13,18 @@
 // ----------------------------------------------------------------------------------
 
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
-using System.Management.Automation;
 using Microsoft.Azure.Management.EdgeGateway;
+using Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models;
 using Microsoft.WindowsAzure.Commands.Utilities.Common;
-using PSResourceModel = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeBandWidthSchedule;
+using System.Management.Automation;
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Bandwidths/DataBoxEdgeBandwidthScheduleRemoveCmdlet.cs]: etter not to make unnecessary changes like the order of using statements.

File: src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeShare.cs [hunk 10]
```diff
@@ -15,7 +16,6 @@ public class PSDataBoxEdgeShare
             ScriptBlock = "$_.share.DataPolicy")]
         [Ps1Xml(Label = "DataFormat", Target = ViewControl.Table,
             ScriptBlock = "$_.share.AzureContainerInfo.DataFormat")]
-
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeShare.cs]: white spaces

File: src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeShare.cs [hunk 11]
```diff
@@ -26,9 +26,13 @@ public class PSDataBoxEdgeShare
 
         [Ps1Xml(Label = "DeviceName", Target = ViewControl.Table)]
         public string DeviceName;
+
```
Comment (VeryEarly) [inline src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeShare.cs]: is this an extra line?

**Comment by VeryEarly** [comment]
Hi @raviten ,

can you also fix these minor issue, like white spaces and extra line.
I'll merge your pr once those were fixed.

File: src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeShare.cs (hunk 11 above)
Comment (raviten) [reply src/DataBoxEdge/DataBoxEdge/Models/PSDataBoxEdgeShare.cs]: Yes

**Comment by VeryEarly** [comment]
Hi @raviten ,

since your credential for test was already in key vault. is it possible to remove live only for these test cases

**Comment by isra-fel** [comment]
Kindly ping @raviten , please remove the live only tests, and fix the conflicts, thanks

**Comment by raviten** [comment]
Will close this PR as this is old one and lot of conflicts, 
Restructuring code is the main reason for this PR, as there is one more update with respect to updates
